### PR TITLE
possibility to delete timestamp entries

### PR DESCRIPTION
**Pull request by Marchuck**
history screen allows to delete timestamp entries

possibility to edit / add before or add timestamp after specified will be added soon(if needed)

### CODE REVIEW

File: app/src/main/java/org/openhdp/hdt/ui/history/HistoryFragment.kt [hunk 1]
```diff
@@ -95,20 +97,29 @@ class HistoryFragment : Fragment() {
...
+        groupieSection.update(state.items.map {
+            renderer.invoke(it)
+        })
+        state.tiemstampToEdit?.let { entry ->
+            AlertDialog.Builder(requireContext())
+                .setTitle(entry.stopwatchName)
+                .setMessage("Are you sure to delete this timestamp?")
+                .setPositiveButton("YES") { dialog, which ->
+                    viewModel.onDelete(entry)
+                }.setNegativeButton("NO") { _, _ -> }
```
Comment (KrzysiekDev) [inline app/src/main/java/org/openhdp/hdt/ui/history/HistoryFragment.kt]: I think that something with the negative button is wrong.

**Review by KrzysiekDev — CHANGES_REQUESTED** [review]
Looking good, but something is wrong with "No" option in delete dialog. ViewModel persist state of showing dialog, because after clicking "No" in delete dialog, when I click on expand the day the dialog is show again. Maybe we should use channel like in this video: https://www.youtube.com/watch?v=6v8iJDJdtMc

File: app/src/main/java/org/openhdp/hdt/ui/history/HistoryFragment.kt (hunk 1 above)
Comment (Marchuck) [reply app/src/main/java/org/openhdp/hdt/ui/history/HistoryFragment.kt]: there was failing test, that's why circle-ci failed

`.setNegativeButton("NO") { _, _ -> }` works the same as
`.setNegativeButton("NO") { dialog, which -> }`

**Review by KrzysiekDev — CHANGES_REQUESTED** [review]
There is still this problem: ViewModel persist state of showing dialog, because after clicking "No" in delete dialog, when I click on expand the day the dialog is show again. Maybe we should use channel like in this video: https://www.youtube.com/watch?v=6v8iJDJdtMc
More information on Discord message.

**Review by KrzysiekDev — APPROVED** [review]
Now everything is OK.

**Pull request merged**